### PR TITLE
Allow asymmetrical rotation limits in `pvlib.tracking.singleaxis`

### LINKED ISSUE #1777: Add `min_angle` argument to `tracking.singleaxis`
In `tracking.singleaxis` the minimum angle of the tracker is assumed to be opposite of the maximum angle, although in some cases the minimum angle could be different. NREL SAM doesn't support that but PVsyst does.

In order to support non symmetrical limiting angles, `tracking.singleaxis` should have another, optional, input, `min_angle`. By default, if not supplied (i.e. value is `None`), the current behavior (`min_angle = -max_angle`) would apply.

Can I propose a PR for this, with modifications to `tracking.singleaxis`, `tracking.SingleAxisTracker` and to `pvsystem.SingleAxisTrackerMount` + corresponding tests?

### PR DESCRIPTION
**Pull request by MichalArieli**
- [x] Closes #1777 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

**Modify max_angle to accept tuples in addition to single values**
(issue #1777)

In `tracking.singleaxis` the minimum angle of the tracker is always the opposite of the maximum angle. However, there are cases where the minimum angle can differ. (While NREL SAM does not support this feature, PVsyst does).

To address this limitation and support non-symmetrical limiting angles, I have made a modification to the `min_angle` parameter in `tracking.singleaxis`. It now accepts tuples of the form (`min_angle`, `max_angle`). To maintain compatibility with the more common symmetrical limits, the parameter still allows a single value.

Please review the changes. Thank you!

### CODE REVIEW

**Comment by cwhanse** [comment]
@MichalArieli I have no idea why GH is saying there are conflicts. Can you merge the current main from pvlib and push again?

**Comment by MichalArieli** [comment]
> @MichalArieli I have no idea why GH is saying there are conflicts. Can you merge the current main from pvlib and push again?

I did it. Are there still conflicts?

File: pvlib/tracking.py [hunk 1]
```diff
@@ -44,12 +44,17 @@ def singleaxis(apparent_zenith, apparent_azimuth,
...
         A value denoting the maximum rotation angle, in decimal degrees,
         of the one-axis tracker from its horizontal position (horizontal
-        if axis_tilt = 0). A max_angle of 90 degrees allows the tracker
-        to rotate to a vertical position to point the panel towards a
-        horizon. max_angle of 180 degrees allows for full rotation.
+        if axis_tilt = 0). If a float is provided, it represents the maximum
+        rotation angle, and the minimum rotation angle is assumed to be the
+        opposite of the maximum angle. If a tuple of (min_angle, max_angle)
+        is provided, it represents both the minimum and maximum rotation angles.
+
```
Comment (cwhanse) [inline pvlib/tracking.py:54]: I feel like we need to explain the orientation that distinguishes `min_angle` from `max_angle`. Suggest adding a new paragraph like:

`A rotation to ``'max_angle'`` is a counter-clockwise rotation about the y-axis of the tracker coordinate system. For example, for a tracker with ``'axis_azimuth'`` oriented to the south, a rotation to ``'max_angle'`` is towards the west, and a rotation toward ``'min_angle'`` is in the opposite direction, toward the east.

**Review by cwhanse — COMMENTED** [review]
@MichalArieli please add a comment to docs/sphinx/source/whatsnew/v0.10.2.rst, and yourself to the list of contributors.

Thanks again for the contribution.

File: pvlib/tracking.py (hunk 1 above)
Comment (MichalArieli) [reply pvlib/tracking.py:54]: Thank you for this input! As you suggested, I have added this paragraph.

**Review by cwhanse — APPROVED** [review]
Thanks @MichalArieli

File: pvlib/tracking.py [hunk 2]
```diff
@@ -190,7 +198,17 @@ def singleaxis(apparent_zenith, apparent_azimuth,
 
     # NOTE: max_angle defined relative to zero-point rotation, not the
     # system-plane normal
-    tracker_theta = np.clip(tracker_theta, -max_angle, max_angle)
+
+
+    # Determine minimum and maximum rotation angles for the tracker based on max_angle.
+    # If max_angle is a single value, assume min_angle is the negative of max_angle.
+    if np.array(max_angle).size == 1:
```
Comment (kandersolar) [inline pvlib/tracking.py:205]:
```suggestion
    if np.isscalar(max_angle):
```
Maybe a bit easier to read while achieving the same thing?

File: docs/sphinx/source/whatsnew/v0.10.2.rst [hunk 3]
```diff
@@ -15,7 +15,9 @@ Enhancements
   :py:func:`pvlib.iotools.get_pvgis_hourly`, :py:func:`pvlib.iotools.get_cams`,
   :py:func:`pvlib.iotools.get_bsrn`, and :py:func:`pvlib.iotools.read_midc_raw_data_from_nrel`.
   (:pull:`1800`)
-
+* Added support for asymmetric limiting angles in :py:func:`pvlib.tracking.singleaxis`, by modifying 
+  the `min_angle` parameter to accept tuples. 
+  (:pull:`1809`)
```
Comment (kandersolar) [inline docs/sphinx/source/whatsnew/v0.10.2.rst:20]:
```suggestion
* Added support for asymmetric limiting angles in :py:func:`pvlib.tracking.singleaxis` and
  :py:class:`pvlib.pvsystem.SingleAxisTrackerMount` by modifying the ``max_angle``
  parameter to accept tuples.  (:issue:`1777`, :pull:`1809`)
```

**Comment by adriesse** [comment]
The title of this PR is quite interesting...

**Comment by kandersolar** [comment]
Superseded by #1852.  Thanks again @MichalArieli!

**Pull request closed without merging**